### PR TITLE
SorrTask485_Unit_test_plot_spectrum

### PR DESCRIPTION
**Pull request by hhxjqm**
#485

### CODE REVIEW

File: core/plotting/test/test_plots.py [hunk 1]
```diff
@@ -128,3 +141,24 @@ def test_plot_effective_correlation_rank2(self) -> None:
...
+
+
+    def test_plot_spectrum2(self) -> None:
+        """
+        Smoke test for `plot_spectrum`.
+
+        - `axes` is list of Matplotlib
+        """
+        test_df = self.get_plot_spectrum1()
+        _, axes = plt.subplots(2, 2, figsize=(10, 8))
```
Comment (DanilYachmenev) [inline core/plotting/test/test_plots.py:162]: it seems like `figsize` is being set over and over through the test methods
lets set it in the `setUpClass` 
as like
```
cls._figsize = (20, 10)
```

and then in every method where it is called replace the value with `self._figsize`
as like with `self._seed` basically

File: core/plotting/test/test_plots.py [hunk 2]
```diff
@@ -128,3 +141,24 @@ def test_plot_effective_correlation_rank2(self) -> None:
...
+        """
+        test_df = self.get_plot_spectrum1()
+        cplmiplo.plot_spectrum(test_df)
+
+
+    def test_plot_spectrum2(self) -> None:
+        """
+        Smoke test for `plot_spectrum`.
+
+        - `axes` is list of Matplotlib
```
Comment (DanilYachmenev) [inline core/plotting/test/test_plots.py:159]: ->
```
- `axes` is a list of Matplotlib axes
```

File: core/plotting/notebooks/gallery_notebook.py [hunk 3]
```diff
@@ -102,4 +103,19 @@
 q_values = np.random.uniform(1, 10, num_q_values).tolist()
 cplocorr.plot_effective_correlation_rank(test_df, q_values)
 
+# %% [markdown]
+# ## `plot_spectrum()`
+
+# %%
+# Set inputs
```
Comment (DanilYachmenev) [inline core/plotting/notebooks/gallery_notebook.py:110]: end comments with dots `.`

File: core/plotting/test/test_plots.py [hunk 4]
```diff
@@ -128,3 +141,24 @@ def test_plot_effective_correlation_rank2(self) -> None:
...
+        """
+        test_df = self.get_plot_spectrum1()
+        _, axes = plt.subplots(2, 2, figsize=(10, 8))
+        axes_flat = axes.flatten()
```
Comment (DanilYachmenev) [inline core/plotting/test/test_plots.py:163]: why do we need to flatten?

**Review by DanilYachmenev — COMMENTED** [review]
all good, lets refactor a bit and done

File: core/plotting/test/test_plots.py (hunk 4 above)
Comment (hhxjqm) [reply core/plotting/test/test_plots.py:163]: axes is an np array
```bash
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In [8], line 4
      2 _, axes = plt.subplots(2, 2, figsize=figsize)
      3 axes_flat = axes.flatten()
----> 4 cplmiplo.plot_spectrum(signal=test_df, axes=axes)

File /app/core/plotting/misc_plotting.py:538, in plot_spectrum(signal, nan_mode, title_prefix, axes)
    536 ax1 = axis_pair[0]
    537 f_pxx, Pxx = sp.signal.welch(data)
--> 538 ax1.semilogy(f_pxx, Pxx)
    539 ax1.set_title(title_prefix + f"{col} power spectral density")
    540 # TODO(*): Maybe put labels on a shared axis.
    541 # ax1.set_xlabel("Frequency")
    542 # ax1.set_ylabel("Power")

AttributeError: 'numpy.ndarray' object has no attribute 'semilogy'
```

File: core/plotting/test/test_plots.py [hunk 5]
```diff
@@ -57,6 +70,8 @@ def setUpClass(cls):
         # Set a specific random seed for the entire test class.
         cls._seed = 42
         np.random.seed(42)
+        # Set a specific figure size
```
Comment (DanilYachmenev) [inline core/plotting/test/test_plots.py:73]: end comments with dots

File: core/plotting/test/test_plots.py [hunk 6]
```diff
@@ -85,9 +100,8 @@ def test_plot_histograms_and_lagged_scatterplot1(self) -> None:
         srs = self.get_plot_histograms_and_lagged_scatterplot1()
         lag = 7
         # TODO(Dan): Remove after integration with `cmamp`. Changes from Cm #4722 are not in `sorrentum` yet.
```
Comment (DanilYachmenev) [inline core/plotting/test/test_plots.py:102]: remove todo

File: core/plotting/test/test_plots.py [hunk 7]
```diff
@@ -106,8 +120,7 @@ def test_plot_heatmap1(self) -> None:
         # TODO(Dan): Move to the notebook config.
```
Comment (DanilYachmenev) [inline core/plotting/test/test_plots.py:120]: remove todo

File: core/plotting/test/test_plots.py [hunk 8]
```diff
@@ -85,9 +100,8 @@ def test_plot_histograms_and_lagged_scatterplot1(self) -> None:
         srs = self.get_plot_histograms_and_lagged_scatterplot1()
         lag = 7
         # TODO(Dan): Remove after integration with `cmamp`. Changes from Cm #4722 are not in `sorrentum` yet.
-        figsize = (20, 20)
         # Plot.
-        cpvistte.plot_histograms_and_lagged_scatterplot(srs, lag, figsize=figsize)
+        cpvistte.plot_histograms_and_lagged_scatterplot(srs, lag, figsize=self._figsize)
```
Comment (DanilYachmenev) [inline core/plotting/test/test_plots.py:104]: run linter of the file

**Review by DanilYachmenev — COMMENTED** [review]
couple nits and ready

**Review by DanilYachmenev — COMMENTED** [review]
lgtm

**Pull request merged**